apps/site/next-data/generators/releaseData.mjs

```diff
@@ -27,47 +31,60 @@ const getNodeReleaseStatus = (now, support) => {
-        status,
-        major: latestVersion.semver.major,
-        version: latestVersion.semver.raw,
-        versionWithPrefix: `v${latestVersion.semver.raw}`,
-        codename: major.support.codename || '',
-        isLts: status === 'LTS',
-        npm: latestVersion.dependencies.npm || '',
-        v8: latestVersion.dependencies.v8 || '',
-        releaseDate: latestVersion.releaseDate || '',
-        modules: latestVersion.modules.version || '',
-      };
-    });
-
-    // nodevu returns duplicated v0.x versions (v0.12, v0.10, ...).
-    // This behavior seems intentional as the case is hardcoded in nodevu,
-    // see https://github.com/cutenode/nodevu/blob/0c8538c70195fb7181e0a4d1eeb6a28e8ed95698/core/index.js#L24.
-    // This line ignores those duplicated versions and takes the latest
-    // v0.x version (v0.12.18). It is also consistent with the legacy
-    // nodejs.org implementation.
-    return nodeReleases.filter(r => r.major !== 0 || r.version === '0.12.18');
+const generateReleaseData = async () => {
+  const nodevuOutput = await nodevu({ fetch: fetch });
+
+  const majors = Object.entries(nodevuOutput).filter(
+    ([version, { support }]) => {
+      // Filter out those without documented support
+      // Basically those not in schedule.json
+      if (!support) {
+        return false;
+      }
+
+      // nodevu returns duplicated v0.x versions (v0.12, v0.10, ...).
+      // This behavior seems intentional as the case is hardcoded in nodevu,
+      // see https://github.com/cutenode/nodevu/blob/0c8538c70195fb7181e0a4d1eeb6a28e8ed95698/core/index.js#L24.
+      // This line ignores those duplicated versions and takes the latest
+      // v0.x version (v0.12.18). It is also consistent with the legacy
+      // nodejs.org implementation.
+      if (version.startsWith('v0.') && version !== 'v0.12') {
+        return false;
+      }
+
+      return true;
+    }
+  );
+
+  return majors.map(([, major]) => {
+    const [latestVersion] = Object.values(major.releases);
+
+    const support = {
+      currentStart: major.support.phases.dates.start,
+      ltsStart: major.support.phases.dates.lts,
+      maintenanceStart: major.support.phases.dates.maintenance,
+      endOfLife: major.support.phases.dates.end,
+    };
+
+    // Get the major release status based on our Release Schedule
+    const status = getNodeReleaseStatus(new Date(), support);
+
+    return {
+      ...support,
+      status,
+      major: latestVersion.semver.major,
+      version: latestVersion.semver.raw,
+      versionWithPrefix: `v${latestVersion.semver.raw}`,
+      codename: major.support.codename || '',
+      isLts: status === 'LTS',
+      npm: latestVersion.dependencies.npm || '',
+      v8: latestVersion.dependencies.v8 || '',
+      releaseDate: latestVersion.releaseDate || '',
+      modules: latestVersion.modules.version || '',
+      minorVersions: Object.entries(major.releases).map(
+        ([, release]) => release.semver.raw
```

Comment:
I actually now wonder if we want to store the Release date too? Minor versions could be an object with the key being the version and value the date? But debatable if we need that info...

Reply:
Yes, I changed the format. It's still an array, but items are objects containing version and release date.